src/html-legend.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * htmlLegend is a attachable widget that can be added to other dc charts to render horizontal/vertical legend
+ * labels.
+ *
+ * Examples:
+ * - {@link http://dc-js.github.com/dc.js/ Nasdaq 100 Index}
+ * @class legend
+ * @memberof dc
+ * @example
+ * chart.legend(dc.htmlLegend().container(legendContainerElement).horizontal(false))
+ * @returns {dc.htmlLegend}
+ */
+dc.htmlLegend = function () {
+    var _legend = {},
+        _parent,
+        _container,
+        _legendText = dc.pluck('name'),
+        _maxItems,
+        _horizontal = false;
+
+    var _l;
+
+    _legend.parent = function (p) {
+        if (!arguments.length) {
+            return _parent;
+        }
+        _parent = p;
+        return _legend;
+    };
+
+    _legend.render = function () {
+        var orientation = _horizontal ? 'horizontal' : 'vertical';
+        _container.select('div.dc-html-legend').remove();
+        _l = _container.append('div')
+            .attr('class', 'dc-html-legend');
+        var legendables = _parent.legendables();
+        if (_maxItems !== undefined) {
+            legendables = legendables.slice(0, _maxItems);
+        }
+        var itemEnter = _l.selectAll('div.dc-legend-item-' + orientation)
+            .data(legendables).enter()
+            .append('div').attr('class', 'dc-legend-item-' + orientation)
+            .on('mouseover', _parent.legendHighlight)
+            .on('mouseout', _parent.legendReset)
+            .on('click', _parent.legendToggle);
+        itemEnter.append('span')
+            .attr('class', 'dc-legend-item-color')
+            .style('background-color', dc.pluck('color'));
+        itemEnter.append('span')
+            .attr('class', 'dc-legend-item-label')
+            .text(_legendText);
+    };
+
+    /**
+     #### .container([selector])
+     Set the container selector for the legend widget. Required.
+     **/
+    _legend.container = function (c) {
+        if (!arguments.length) {
+            return _container;
+        }
+        _container = d3.select(c);
+        return _legend;
+    };
+
+    /**
+     #### .horizontal([boolean])
+     Display the legend horizontally instead of horizontally
```

Comment:
Shouldn't this comment say 'vertically' not 'horizontally' at the end of the sentence?

Reply:
yes, will need to correct that, Thanks for pointing it out.